staging/src/k8s.io/component-helpers/nodedeclaredfeatures/features/registry.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package features
+
+import (
+	"fmt"
+	"regexp"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/component-helpers/nodedeclaredfeatures"
+)
+
+// featureNameRegexp defines the allowed format for feature names.
+// The first segment must be in UpperCamelCase. Subsequent segments (separated by '/')
+// can be in either UpperCamelCase or lowerCamelCase.
+var featureNameRegexp = regexp.MustCompile(`^[A-Z][a-zA-Z0-9]*(\/[a-zA-Z][a-zA-Z0-9]*)*$`)
+
+// AllFeatures is the central registry for all declared features.
+// New features are added to this list to be automatically included in both
+// discovery and inference logic.
+var AllFeatures = []nodedeclaredfeatures.Feature{}
+
+func init() {
+	if err := ValidateFeatures(); err != nil {
```

Comment:
If this is validating the hardcoded list of features, it would be better to put this in a unit test rather than performed on startup.

Reply:
Done.